src/firejail/util.c

```diff
@@ -103,6 +103,38 @@ void errLogExit(char* fmt, ...) {
 	exit(1);
 }
 
+// Returns whether all supplementary groups can be safely dropped
+int check_can_drop_all_groups() {
+	static int can_drop_all_groups = -1;
+
+	// Avoid needlessly checking (and printing) things twice
+	if (can_drop_all_groups != -1)
+		goto out;
+
+	// nvidia cards require video group; ignore nogroups
+	if (access("/dev/nvidiactl", R_OK) == 0 && arg_no3d == 0) {
+		fwarning("NVIDIA card detected, nogroups command ignored\n");
+		can_drop_all_groups = 0;
+	}
+
```

Comment:
I'd move the `else if ...` here and put the comment below it. Now it looks like earlier `if (access(...` didn't have an `else` part but of course it does when you read further.

Reply:
@topimiettinen [on Dec 2](https://github.com/netblue30/firejail/pull/4732#discussion_r761459575):

> I'd move the `else if ...` here and put the comment below it. Now it looks
> like earlier `if (access(...` didn't have an else part but of course it does
> when you read further.

I think I got this idea from something like this code on profile.c:

```C
	// mkdir
	if (strncmp(ptr, "mkdir ", 6) == 0) {
		fs_mkdir(ptr + 6);
		return 1;	// process mkdir again while applying blacklists
	}
	// mkfile
	if (strncmp(ptr, "mkfile ", 7) == 0) {
		fs_mkfile(ptr + 7);
		return 1;	// process mkfile again while applying blacklists
	}
	// sandbox name
	else if (strncmp(ptr, "name ", 5) == 0) {
		cfg.name = ptr + 5;
		if (strlen(cfg.name) == 0) {
			fprintf(stderr, "Error: invalid sandbox name\n");
			exit(1);
		}
		return 0;
	}
	else if (strcmp(ptr, "ipc-namespace") == 0) {
		arg_ipc = 1;
		return 0;
	}
```

But yeah, it's not very obvious; I'll change it along with some other stuff and
force-push.

Reply:
Done.  Just moving the comment made things look a bit crowded, so I added a few
more `goto`s instead.